vscode/microsoft-kiota/src/util.ts

```diff
@@ -66,4 +66,35 @@ export function findAppPackageDirectory(directory: string): string | null {
   }
 
   return null;
+}
+
+export async function findLockFile(directory: string): Promise<string | undefined> {
+  return new Promise((resolve) => {
```

Comment:
consider using glob instead to avoid scanning the whole directory manually.

Reply:
A note on glob, it's shipped with node 22+, below that you need to install the package from npm.

Reply:
turns out vscode has in inbuilt function that uses a glob pattern